src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_pix2pix_zero.py

```diff
@@ -852,7 +915,247 @@ def __call__(
+
+        # 7. Denoising loop where we obtain the cross-attention maps.
+        num_warmup_steps = len(timesteps) - num_inference_steps * self.inverse_scheduler.order
+        with self.progress_bar(total=num_inference_steps - 2) as progress_bar:
+            for i, t in enumerate(timesteps.flip(0)[1:-1]):
+                # expand the latents if we are doing classifier free guidance
+                latent_model_input = torch.cat([latents] * 2) if do_classifier_free_guidance else latents
+                latent_model_input = self.inverse_scheduler.scale_model_input(latent_model_input, t)
+
+                # predict the noise residual
+                noise_pred = self.unet(
+                    latent_model_input,
+                    t,
+                    encoder_hidden_states=prompt_embeds,
+                    cross_attention_kwargs={"timestep": t},
+                ).sample
+
+                # perform guidance
+                if do_classifier_free_guidance:
+                    noise_pred_uncond, noise_pred_text = noise_pred.chunk(2)
+                    noise_pred = noise_pred_uncond + guidance_scale * (noise_pred_text - noise_pred_uncond)
+
+                # regularization of the noise prediction
+                e_t = noise_pred
+                with torch.enable_grad():
+                    for _outer in range(num_reg_steps):
+                        if lambda_ac > 0:
+                            for _inner in range(num_ac_rolls):
+                                _var = torch.autograd.Variable(e_t.detach().clone(), requires_grad=True)
+                                l_ac = self.auto_corr_loss(_var)
+                                l_ac.backward()
+                                _grad = _var.grad.detach() / num_ac_rolls
+                                e_t = e_t - lambda_ac * _grad
+                        if lambda_kl > 0:
+                            _var = torch.autograd.Variable(e_t.detach().clone(), requires_grad=True)
+                            l_kld = self.kl_divergence(_var)
+                            l_kld.backward()
+                            _grad = _var.grad.detach()
+                            e_t = e_t - lambda_kl * _grad
+                        e_t = e_t.detach()
+
+                noise_pred = e_t
+
+                # compute the previous noisy sample x_t -> x_t-1
+                latents = self.inverse_scheduler.step(
+                    noise_pred, t, latents, reverse=True, **extra_step_kwargs
+                ).prev_sample
+
+                # call the callback, if provided
+                if i == len(timesteps) - 1 or (
+                    (i + 1) > num_warmup_steps and (i + 1) % self.inverse_scheduler.order == 0
+                ):
+                    progress_bar.update()
+                    if callback is not None and i % callback_steps == 0:
+                        callback(i, t, latents)
+
+        x_inv = latents.detach().clone()
+        # reconstruct the image
```

Comment:
(nit): needs to come above the code line and we're reconstructing the noise here and not the image.